src/gt4py/next/ffront/foast_passes/type_deduction.py

```diff
@@ -556,7 +579,7 @@ def visit_Call(self, node: foast.Call, **kwargs) -> foast.Call:
             )
         except GTTypeError as err:
             raise FieldOperatorTypeDeductionError.from_foast_node(
-                node, msg=f"Invalid argument types in call to `{node.func.id}`!"
+                node, msg=f"Invalid argument types in call to `{func_str_repr}`!"
```

Comment:
Once you have implemented the change that you wanted to make to avoid func_str_repr, it should be good to go.